### ISSUE TITLE
Error with pre-commit installation

### ISSUE THREAD

**Issue by MRaza98** [sqlfluff/sqlfluff#5623]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I added sqlfluff to pre-commit hooks and this is the error I get:

An unexpected error has occurred: CalledProcessError: command: ('C:\\Program Files\\Git\\cmd\\git.EXE', 'checkout', 'v2.3.5')
return code: 1
stdout: (none)
stderr:
    error: pathspec 'v2.3.5' did not match any file(s) known to git
Check the log at C:\Users\muhammad.raza\.cache\pre-commit\pre-commit.log

Here is my configuration of the hook:

![image](https://github.com/sqlfluff/sqlfluff/assets/49342984/2ee9a0eb-921b-4717-85f3-2a42848784ae)

### Expected Behaviour

I would expect sqlfluff to work.

### Observed Behaviour

I get the error: 

[INFO] Initializing environment for https://github.com/sqlfluff/sqlfluff.git.
An unexpected error has occurred: CalledProcessError: command: ('C:\\Program Files\\Git\\cmd\\git.EXE', 'checkout', 'v2.3.5')
return code: 1
stdout: (none)
stderr:
    error: pathspec 'v2.3.5' did not match any file(s) known to git

### How to reproduce

Create a basic .pre-commit-config.yaml file and add the hooks like in screenshot provided.

### Dialect

Not sure

### Version

2.3.5

### Configuration

I get this error when I run sqlfluff config init:

sqlfluff : The term 'sqlfluff' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, 
verify that the path is correct and try again.
At line:1 char:1
+ sqlfluff config init
+ ~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (sqlfluff:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
Your `rev` line should be `2.3.5` instead of `v2.3.5`. The rev line should be either a [tag](https://github.com/sqlfluff/sqlfluff/tags) or a commit ID. In this case, SQLFluff doesn't use the `v` prefix in tags.

```yaml
default_stages: [commit, push]
default_language_version:
  python: python3
minimum_pre_commit_version: "1.20.0"
repos:
  - repo: https://github.com/sqlfluff/sqlfluff.git
    rev: 2.3.5
    hooks:
      - id: sqlfluff-lint
      - id: sqlfluff-fix
```
